clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -11252,6 +11252,12 @@ llvm::Type *CommonSPIRTargetCodeGenInfo::getOpenCLType(CodeGenModule &CGM,
 #define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
     case BuiltinType::Id:                                                      \
       return getSPIRVImageType(Ctx, "spirv.Image", #ImgType, AQ_##Suffix);
+#include "clang/Basic/OpenCLImageTypes.def"
+#define IMAGE_TYPE(ImgType, Id, SingletonId, Access, Suffix)                   \
```

Comment:
Shouldn't we do that in the community?

Reply:
The `Sampled*` types haven't been upstreamed yet, which means this change itself can't be upstreamed.